napi_basics.md

```diff
@@ -0,0 +1,139 @@
+# N-API - Basics
+
+N-API is an API exposed by Node to developers that allow them to author native 
+add-ons in an ABI-stable manner. This means that these add-ons can work 
+against future versions of Node without recompilation, and developers get the 
+additional benefit that their add-ons can work independently of the underlying 
+JavaScript Virtual Machine being used by Node.
+
+APIs exposed by the N-API surface are generally used to create and manipulate
+JavaScript values. Concepts and operations generally map to ideas specified 
+in the ECMA262 Language Specification. The APIs generally have the following 
+properties:
+- All N-API calls generally return a status code of type `napi_status`. This
+  status can be used to determine whether the API call succeeded or failed.
+- The actual value being returned by the API is returned in an out parameter.
+- All JavaScript values are abstracted behind an opaque type named 
+  `napi_value`.
+- In case of an error status code, additional information can be obtained 
+  using `napi_get_last_error_info`. More information can be found in the error
+  handling section (<PLACEHOLDER: link to error handling doc>)
+
+## N-API General types
+
+N-API exposes the following fundamental datatypes as abstractions that are 
+consumed by the various APIs. These APIs should be treated as opaque types 
```

Comment:
"These types should be treated as opaque, introspectable only with other N-API calls."